tests/urlpatterns_reverse/tests.py

```diff
@@ -591,251 +608,362 @@ def test_redirect_view_object(self):
 
 @override_settings(ROOT_URLCONF='urlpatterns_reverse.namespace_urls')
 class NamespaceTests(SimpleTestCase):
-
     def test_ambiguous_object(self):
-        "Names deployed via dynamic URL objects that require namespaces can't be resolved"
-        with self.assertRaises(NoReverseMatch):
-            reverse('urlobject-view')
-        with self.assertRaises(NoReverseMatch):
-            reverse('urlobject-view', args=[37, 42])
-        with self.assertRaises(NoReverseMatch):
-            reverse('urlobject-view', kwargs={'arg1': 42, 'arg2': 37})
+        """
+        Names deployed via dynamic URL objects that require namespaces can't be resolved.
+        """
+        test_urls = [
+            ('urlobject-view', [], {}, None),
+            ('urlobject-view', [37, 42], {}, None),
+            ('urlobject-view', [], {'arg1': 42, 'arg2': 37}, None),
+        ]
+
+        for name, args, kwargs, current_app in test_urls:
+            with self.subTest(name=name, args=args, kwargs=kwargs, current_app=current_app):
```

Comment:
Is there a need to add `current_app=current_app` when it's always None for many of these tests.